### ISSUE TITLE
Version 0.4.0 does not output file.

### ISSUE THREAD

**Issue by hamiltop** [madsflensted/elm-brunch#8]
Is there some documentation I'm missing?

If I bump my version to 0.4.0, then everything breaks.

**Comment by madsflensted** [comment]
Can you provide the elm-brunch part of your brunch config file?

0.4 skips non main modules on init.

**Comment by hamiltop** [comment]
```
    elmBrunch: {
      elmFolder: 'web/elm',
      mainModules: ['src/ReviewYourBudget.elm'],
      outputFolder: '../static/vendor/'
    },
```

in `web/elm`:

```
➜  elm git:(master) ✗ ls src
GuidedTour.elm       ReviewYourBudget.elm
➜  elm git:(master) ✗ cat elm-package.json
{
    "version": "1.0.0",
    "summary": "helpful summary of your project, less than 80 characters",
    "repository": "https://github.com/user/project.git",
    "license": "BSD3",
    "source-directories": [
        "src"
    ],
    "exposed-modules": [],
    "dependencies": {
        "elm-lang/core": "3.0.0 <= v < 4.0.0",
        "evancz/elm-html": "4.0.2 <= v < 5.0.0"
    },
    "elm-version": "0.16.0 <= v < 0.17.0"
}
```

output of `brunch b -d` with `0.4.0`:

```
 brunch:file-list Reading 'web/elm/src/GuidedTour.elm' +4ms
  brunch:watch File 'web/elm/src/GuidedTour.elm' received event 'add' +1ms
  brunch:file-list Reading 'web/elm/src/ReviewYourBudget.elm' +1ms
  brunch:watch File 'web/elm/src/ReviewYourBudget.elm' received event 'add' +0ms
  brunch:source-file Initializing fs_utils.SourceFile: {"path":"web/elm/src/GuidedTour.elm","isntModule":false,"isWrapped":true} +3ms
  brunch:source-file Initializing fs_utils.SourceFile: {"path":"web/elm/src/ReviewYourBudget.elm","isntModule":false,"isWrapped":true} +4ms
  brunch:pipeline Compiling 'web/elm/src/GuidedTour.elm' with 'ElmCompiler' +2ms
  brunch:file-list Compiled file 'web/elm/src/GuidedTour.elm'... +2ms
  brunch:pipeline Compiling 'web/elm/src/ReviewYourBudget.elm' with 'ElmCompiler' +0ms
  brunch:file-list Compiled file 'web/elm/src/ReviewYourBudget.elm'... +2ms
```

output of `brunch -b -d` with `0.3.1`:

```
  brunch:file-list Reading 'web/elm/src/GuidedTour.elm' +5ms
  brunch:watch File 'web/elm/src/GuidedTour.elm' received event 'add' +1ms
  brunch:file-list Reading 'web/elm/src/ReviewYourBudget.elm' +0ms
  brunch:watch File 'web/elm/src/ReviewYourBudget.elm' received event 'add' +1ms
  brunch:watch File 'deps/phoenix/web/static/js' received event 'addDir' +1ms
  brunch:watch File 'deps/phoenix_html/web/static/js' received event 'addDir' +0ms
  brunch:source-file Initializing fs_utils.SourceFile: {"path":"web/elm/src/GuidedTour.elm","isntModule":false,"isWrapped":true} +3ms
  brunch:source-file Initializing fs_utils.SourceFile: {"path":"web/elm/src/ReviewYourBudget.elm","isntModule":false,"isWrapped":true} +3ms
  brunch:pipeline Compiling 'web/elm/src/GuidedTour.elm' with 'ElmCompiler' +2ms
Elm compile: src/ReviewYourBudget.elm, in web/elm, to ../static/vendor/reviewyourbudget.js
  brunch:pipeline Compiling 'web/elm/src/ReviewYourBudget.elm' with 'ElmCompiler' +9ms
Elm compile: src/ReviewYourBudget.elm, in web/elm, to ../static/vendor/reviewyourbudget.js
```

**Comment by madsflensted** [comment]
Ok I spotted the problem,  it relates to difference between the filepath
used by brunch and the one used by Elm. It has been there a while but the
latest change made it visible.

Thanks for reporting it, I will take a look at fixing.

```
elmBrunch: {
  elmFolder: 'web/elm',
  mainModules: ['src/ReviewYourBudget.elm'],
  outputFolder: '../static/vendor/'
},
```

in web/elm:

➜  elm git:(master) ✗ ls src
GuidedTour.elm       ReviewYourBudget.elm
➜  elm git:(master) ✗ cat elm-package.json
{
    "version": "1.0.0",
    "summary": "helpful summary of your project, less than 80 characters",
    "repository": "https://github.com/user/project.git",
    "license": "BSD3",
    "source-directories": [
        "src"
    ],
    "exposed-modules": [],
    "dependencies": {
        "elm-lang/core": "3.0.0 <= v < 4.0.0",
        "evancz/elm-html": "4.0.2 <= v < 5.0.0"
    },
    "elm-version": "0.16.0 <= v < 0.17.0"
}

output of brunch b -d with 0.4.0:

 brunch:file-list Reading 'web/elm/src/GuidedTour.elm' +4ms
  brunch:watch File 'web/elm/src/GuidedTour.elm' received event 'add' +1ms
  brunch:file-list Reading 'web/elm/src/ReviewYourBudget.elm' +1ms
  brunch:watch File 'web/elm/src/ReviewYourBudget.elm' received event 'add' +0ms
  brunch:source-file Initializing fs_utils.SourceFile:
{"path":"web/elm/src/GuidedTour.elm","isntModule":false,"isWrapped":true}
+3ms
  brunch:source-file Initializing fs_utils.SourceFile:
{"path":"web/elm/src/ReviewYourBudget.elm","isntModule":false,"isWrapped":true}
+4ms
  brunch:pipeline Compiling 'web/elm/src/GuidedTour.elm' with 'ElmCompiler' +2ms
  brunch:file-list Compiled file 'web/elm/src/GuidedTour.elm'... +2ms
  brunch:pipeline Compiling 'web/elm/src/ReviewYourBudget.elm' with
'ElmCompiler' +0ms
  brunch:file-list Compiled file 'web/elm/src/ReviewYourBudget.elm'... +2ms

output of brunch -b -d with 0.3.1:

  brunch:file-list Reading 'web/elm/src/GuidedTour.elm' +5ms
  brunch:watch File 'web/elm/src/GuidedTour.elm' received event 'add' +1ms
  brunch:file-list Reading 'web/elm/src/ReviewYourBudget.elm' +0ms
  brunch:watch File 'web/elm/src/ReviewYourBudget.elm' received event 'add' +1ms
  brunch:watch File 'deps/phoenix/web/static/js' received event 'addDir' +1ms
  brunch:watch File 'deps/phoenix_html/web/static/js' received event
'addDir' +0ms
  brunch:source-file Initializing fs_utils.SourceFile:
{"path":"web/elm/src/GuidedTour.elm","isntModule":false,"isWrapped":true}
+3ms
  brunch:source-file Initializing fs_utils.SourceFile:
{"path":"web/elm/src/ReviewYourBudget.elm","isntModule":false,"isWrapped":true}
+3ms
  brunch:pipeline Compiling 'web/elm/src/GuidedTour.elm' with 'ElmCompiler' +2ms
Elm compile: src/ReviewYourBudget.elm, in web/elm, to
../static/vendor/reviewyourbudget.js
  brunch:pipeline Compiling 'web/elm/src/ReviewYourBudget.elm' with
'ElmCompiler' +9ms
Elm compile: src/ReviewYourBudget.elm, in web/elm, to
../static/vendor/reviewyourbudget.js

**Comment by madsflensted** [comment]
Ok - I have pushed a fix and published version 0.4.1 on npm.

@hamiltop Please give it a try when you get a chance.

**Comment by hamiltop** [comment]
Looks good.
